### ISSUE TITLE
Incorrect number of dimensions" error when using model="arima"

### ISSUE THREAD

**Issue by ldlpdx** [IQSS/Zelig#145]
Hello,

I posted this on the Google forum yesterday, but got no response. So I'm hoping that whoever maintains this package is checking these issues.

I'm getting an error when I try to run an `arima` model with the zelig package. I'm using MI data that were created with Amelia. Here is a short summary of my id and response variables:

```
 $ imp20:'data.frame':  442 obs. of  50 variables:
  ..$ region                     : Factor w/ 4 levels "Central Africa",..: 3 3 3 3 3 3 3 3 3 3 ...
  ..$ subregionid                : Factor w/ 4 levels "FC","FE","FS",..: 3 3 3 3 3 3 3 3 3 3 ...
  ..$ country                    : Factor w/ 34 levels "Angola","Benin",..: 1 1 1 1 1 1 1 1 1 1 ...
  ..$ ISO2                       : Factor w/ 34 levels "AO","BF","BJ",..: 1 1 1 1 1 1 1 1 1 1 ...
  ..$ ISO3                       : Factor w/ 34 levels "AGO","BEN","BFA",..: 1 1 1 1 1 1 1 1 1 1 ...
  ..$ year                       : num [1:442] 2002 2003 2004 2005 2006 ...
  ....
  ..$ Diff.from.expected         : num [1:442] -0.0415 -0.0415 -0.0415 -0.0415 -0.0415 ...
```

`Diff.from.expected` is my response variable. And here is the code that I've run along with the error I'm getting.

```
> z1$zelig(Diff.from.expected~GNI, order=c(1,0,1), model="arima",
+                data = a.coVarsTrans.more, ts="year", cs="country")
Error in data[, cs] : incorrect number of dimensions
```

So it appears to me that there is an issue with the `cs='country'` call, but I'm not sure what the issue is. 

Also, there is no explanation in the documentation what the `order=` call is and how to determine its parameters, so I just used what was in the documentation as a ts/cs example. No idea if this was correct.

Any help with these questions would be much appreciated. Thanks in advance!

**Comment by cchoirat** [comment]
``order`` corresponds to http://stats.stackexchange.com/questions/44992/what-are-the-values-p-d-q-in-arima.

It would be easier for us to help if you could provide a replicable and self-contained example.

**Comment by ldlathrop** [comment]
Thanks, @cchoirat. Here is the link to my [saved Amelia .Rdata file](https://github.com/ldlathrop/Elephant-data-analysis/blob/master/coVarsTrans.RData). There are 20 imputations.

And here are a couple of examples I've tried to run:

    ## Zelig models
    library(Zelig)
    library(mitools)
    
    z1 <- zarima$new()
    z1$zelig(Diff.from.expected~HDI, order=c(1,0,1), model="arima",
               data = a.coVarsTrans.more, ts="year", cs="country")

    z1$zelig(Diff.from.expected~HDI+GNI, order=c(1,0,1), model="arima",
               data = a.coVarsTrans.more, ts="year", cs="country")
    
    z3 <- zarima$new()
    z3$zelig(PIKE.regional~Armed.Conflict+Non.State.Conflict, order=c(1,0,1), model="arima",
         data = a.coVarsTrans.more, ts="year", cs="country")

All have the same error `Error in data[, cs] : incorrect number of dimensions`.

And again, it would be very helpful if you could please explain to me what is meant by the `order` call and how to determine its parameters, as there is no explanation in the documentation, and I have been unable to find an explanation in the Google results searching for time-series, cross-sectional analysis.

I hope this is helpful. Thanks!

**Comment by ldlathrop** [comment]
**UPDATE:** I did finally find an answer about the `order` call, but I still need help with the `incorrect number of dimensions` error. Thanks!

**Comment by christophergandrud** [comment]
The root problem seems to be with your dependent variable. It doesn't vary for a number of countries (e.g. Senegal, Togo) and so `arima` is unable to model it.

![imp1](https://cloud.githubusercontent.com/assets/1285805/22425718/7ba9f050-e6ca-11e6-85aa-d3c8da53171d.png)


For example, if we look at your first imputed data set only:

```R
one_df <- a.coVarsTrans.more$imputations$imp1
``` 

and run the model only for Angola and Benin (for which the DV does vary):

```R
one_df_sub <- subset(one_df, country %in% 
                         c("Angola", "Benin"))
z1 <- zarima$new()
z1$zelig(Diff.from.expected ~ HDI, order = c(1, 0, 1),
         data = one_df_sub, ts = "year", cs = "country")
```

There is no error.

If we add in Senegal:

```R
one_df_sub <- subset(one_df, country %in% 
                          c("Angola", "Benin", "Senegal"))
z1 <- zarima$new()
z1$zelig(Diff.from.expected ~ HDI, order = c(1, 0, 1),
          data = one_df_sub, ts = "year", cs = "country")

## Error in solve.default(res$hessian * n.used, A) : 
##  Lapack routine dgesv: system is exactly singular: U[3,3] = 0 In addition: Warning message:
## In summary.lm(fit) : essentially perfect fit: summary may be unreliable
```

My hunch is that because `arima` is not returning estimates in the expected form that it later on leads to the `Error in data[, cs] : incorrect number of dimensions`. 

So, Zelig really needs to return an error earlier and be a bit more informative.

**Comment by christophergandrud** [comment]
I've added in the error so that `arima` fails informatively if there is no dependent variable variation for one of the cases: df3e941f92ab66def74eaaf19b86f827f741a76a

That being said. This doesn't actually prevent `Error in data[, cs] : incorrect number of dimensions`. I haven't been able to reproduce this error for other data sets and it occurs before `arima` is even called. 

Actually inserting `browser()` right after the `zelig` method starts (after line [229](https://github.com/IQSS/Zelig/blob/df3e941f92ab66def74eaaf19b86f827f741a76a/R/model-zelig.R#L228)) still gives me the error before `browser` is even hit. 

I'm at a bit of a loss as to how this could happen. Also strange is that in RStudio the Error does not give me a `Show Traceback` option. Any thoughts @tercer?

**Comment by christophergandrud** [comment]
I'm going to close this issue for now.
